src/promptflow/promptflow/_sdk/_service/app.py

```diff
@@ -92,4 +96,22 @@ def log_after_request_info(response):
             )
             return response
 
+        # Start a monitor process using detach mode. It will stop pfs service if no request to pfs service in 1h in
+        # python scenario. For C# scenario, pfs will live until the process is killed manually.
+        def monitor_request():
+            while True:
+                time.sleep(60)
+                # For python scenario, since we start waitress in cli, there will be two app. The one used to log in
+                # the parent process will have no "last_request_time" in app.config since the app doesn't run.
+                if "last_request_time" in app.config and datetime.now() - app.config["last_request_time"] > timedelta(
+                    hours=1
+                ):
+                    port = get_port_from_config()
```

Comment:
Raise a TODO here: check if we have any not complete work? like persist all traces.

Reply:
added